components/ILIAS/Authentication/tests/ilSessionTest.php

```diff
@@ -71,180 +63,279 @@ protected function setGlobalVariable(string $name, $value): void
-                $this->getMockBuilder(ilDBStatement::class)->disableAutoReturnValueGeneration()->getMock(),
-                $this->getMockBuilder(ilDBStatement::class)->disableAutoReturnValueGeneration()->getMock()
-            );
-        $ilDB->expects($this->exactly(4))->method("fetchAssoc")->willReturn(
-            ["data" => "Testdata"],
+        )->willReturnOnConsecutiveCalls(
+            $this->getMockBuilder(ilDBStatement::class)->disableAutoReturnValueGeneration()->getMock(),
+            $this->getMockBuilder(ilDBStatement::class)->disableAutoReturnValueGeneration()->getMock(),
+            $this->getMockBuilder(ilDBStatement::class)->disableAutoReturnValueGeneration()->getMock(),
+            $this->getMockBuilder(ilDBStatement::class)->disableAutoReturnValueGeneration()->getMock(),
+            $this->getMockBuilder(ilDBStatement::class)->disableAutoReturnValueGeneration()->getMock(),
+            $this->getMockBuilder(ilDBStatement::class)->disableAutoReturnValueGeneration()->getMock(),
+            $this->getMockBuilder(ilDBStatement::class)->disableAutoReturnValueGeneration()->getMock(),
+            $this->getMockBuilder(ilDBStatement::class)->disableAutoReturnValueGeneration()->getMock(),
+            $this->getMockBuilder(ilDBStatement::class)->disableAutoReturnValueGeneration()->getMock(),
+            $this->getMockBuilder(ilDBStatement::class)->disableAutoReturnValueGeneration()->getMock(),
+            $this->getMockBuilder(ilDBStatement::class)->disableAutoReturnValueGeneration()->getMock()
+        );
+        $ilDB->expects($this->exactly(4))->method('fetchAssoc')->willReturn(
+            ['data' => 'Testdata'],
             [],
-            ["data" => "Testdata"],
+            ['data' => 'Testdata'],
             []
         );
-        $ilDB->expects($this->exactly(1))->method("fetchObject")->willReturn((object) array(
-            'data' => "Testdata"
-        ));
+        $ilDB->expects($this->once())->method('fetchObject')->willReturn(
+            (object) [
+                'data' => 'Testdata'
+            ]
+        );
 
-        $ilDB->method("manipulate")->withConsecutive(
+        $ilDB->method('manipulate')->withConsecutive(
             ['DELETE FROM usr_sess_istorage WHERE session_id = 123456'],
             ['DELETE FROM usr_session WHERE session_id = e10adc3949ba59abbe56e057f20f883e'],
             ['DELETE FROM usr_session WHERE user_id = e10adc3949ba59abbe56e057f20f883e']
-        )->
-        willReturnOnConsecutiveCalls(
+        )->willReturnOnConsecutiveCalls(
             1,
             1,
             1
         );
 
-        $result = "";
-        ilSession::_writeData("123456", "Testdata");
-        if (ilSession::_exists("123456")) {
-            $result .= "exists-";
+        $cron_manager = $this->createMock(ilCronManager::class);
+        $cron_manager->method('isJobActive')->with($this->isType('string'))->willReturn(false);
```

Comment:
This should return true to my understanding, so the ilSessionTest assumes that the cleanup will be run by Cron and not cause the flacky behavior.